### PR TITLE
Add SIRIUS and support for SIRIUS in CP2K

### PR DESCRIPTION
**Pull request by dev-zero**

### CODE REVIEW

**Comment by dev-zero** [comment]
@ltalirz fyi

**Comment by ltalirz** [comment]
Thanks!

File: var/spack/repos/builtin/packages/libvdwxc/package.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    depends_on("pfft", when="+pfft")
+
+    # pfft needs MPI
+    conflicts("~mpi", "+pfft")
+
+    def configure_args(self):
+        spec = self.spec
+
+        args = [
+            "--{}-pfft".format(
```
[the remark this answers is not in the document]
Comment (dev-zero) [reply var/spack/repos/builtin/packages/libvdwxc/package.py]: done, I wondered why the indices were still all over the place ;-)

File: var/spack/repos/builtin/packages/sirius/package.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+            _def('+elpa'),
+            _def('+vdwxc'),
+            _def('+scalapack'),
+            _def('+fortran', 'CREATE_FORTRAN_BINDINGS'),
+        ]
+
+        if self.spec.satisfies('+elpa'):
+            elpa = self.spec['elpa']
+            elpa_suffix = '_openmp' if elpa.satisfies('+openmp') else ''
+            elpa_incdir = os.path.join(
```
[the remark this answers is not in the document]
Comment (dev-zero) [reply var/spack/repos/builtin/packages/sirius/package.py]: I guess so. ELPA installs the headers in a way such that you can have multiple versions/variants of ELPA installed in parallel on the same system by including the relevant variant (openmp/non-openmp) and the version in the header folder prefix:

```
$prefix/include
  + elpa_openmp-2017.05.003/
     + elpa/
          elpa.h
          elpa_constants.h
          elpa_generated.h
          elpa_generated_legacy.h
          elpa_generic.h
          elpa_legacy.h
```

My guess is that therefore nobody directly writes a `#include <elpa_openmp-2017.05.003/elpa/elpa.h>` in their source, but rather relies on the build system to get from the user either ELPA's variant name+version or directly the include path `$prefix/include/elpa_openmp-2017.05.003`.

Commit follows...

[the remark this answers is not in the document]
Comment (dev-zero) [reply var/spack/repos/builtin/packages/sirius/package.py]: @scheibelp done: 0a6f1ac

**Comment by scheibelp** [comment]
Thanks!

**Pull request merged**